tests/10apidoc/20presence.pl

```diff
@@ -1,15 +1,55 @@
 my $fixture = local_user_fixture();
 
+push our @EXPORT, qw( matrix_get_presence_status matrix_set_presence_status );
+
+=head2 matrix_get_presence_status
+
+   $status = matrix_get_presence_status( $user )
+
+Returns a HASH reference containing the user's presence status. This will
+contain a C<presence> field, and optionally a C<status_msg> field as well if
+the user has one set.
+
+=cut
+
+sub matrix_get_presence_status
+{
+   my ( $user ) = @_;
+
+   do_request_json_for( $user,
+      method => "GET",
+      uri    => "/api/v1/presence/:user_id/status",
+   );
+}
+
+=head2 matrix_set_presence_status
+
+   matrix_set_presence_status( $user, $presence, %params )
+
+Sets the presence status of the given C<$user> to C<$presence>, with optional
+additional parameters (such as C<status_msg>) given in C<%params>.
+
+=cut
+
+sub matrix_set_presence_status
+{
+   my ( $user, $presence, %params ) = @_;
+
+   do_request_json_for( $user,
+      method => "PUT",
+      uri    => "/api/v1/presence/:user_id/status",
+
+      content => { presence => $presence, %params }
+   )->then_done();
```

Comment:
Conceptually a "set" operation shouldn't return a result, so this just throws away the returned JSON body of the HTTP response.